### PR TITLE
Allow `QueryParameter::UnlockableByAddress` for more routes

### LINKED ISSUE #1269: Add unlockableByAddress QueryParameter to alias/basic/nft endpoints
## Description

Add unlockableByAddress QueryParameter to alias/basic/nft endpoints in Rust and bindings

https://github.com/iotaledger/tips/pull/136/commits/0dd6217a49c7940ee1191ab05b09d4fa6cbff86b

## Motivation

Support the same as the indexer

## Are you planning to do it yourself in a pull request?

No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Added `QueryParameter::UnlockableByAddress` to allowed query paramaters for `Client::{alias_output_ids(), basic_output_ids(), nft_output_ids()}` since this is now allowed by the indexer https://github.com/iotaledger/tips/commit/0dd6217a49c7940ee1191ab05b09d4fa6cbff86b

Also changed the syncing in the wallet to make use of it

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/1269

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Running an example with
```Rust
    let output_ids_response = client
        .alias_output_ids([QueryParameter::UnlockableByAddress(address)])
        .await?;
```
and also synced a wallet with
```Rust
SyncOptions {
            nft: NftSyncOptions {
                basic_outputs: true,
                ..Default::default()
            },
            ..Default::default()
        }
```

### CODE REVIEW

File: bindings/nodejs/CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Added `SeedSecretManager` to `SecretManagerType`;
+- Added `UnlockableByAddress` to `AliasQueryParameter, NftQueryParameter, QueryParameter`;
```
Comment (thibault-martinez) [inline bindings/nodejs/CHANGELOG.md]: Not really a fix?

File: sdk/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## 1.1.1 - 2023-MM-DD
 
+### Added
+
+- Added `QueryParameter::UnlockableByAddress` to allowed query paramaters for `Client::{alias_output_ids(), basic_output_ids(), nft_output_ids()}`;
```
Comment (thibault-martinez) [inline sdk/CHANGELOG.md]:
```suggestion
- `QueryParameter::UnlockableByAddress` to allowed query parameters for `Client::{alias_output_ids(), basic_output_ids(), nft_output_ids()}`;
```

**Review by thibault-martinez — COMMENTED** [review]
Is there nowhere we can actually use it already?

File: bindings/nodejs/CHANGELOG.md (hunk 1 above)
Comment (Thoralf-M) [reply bindings/nodejs/CHANGELOG.md]: Changed

**Comment by Thoralf-M** [comment]
> Is there nowhere we can actually use it already?

There is, should I include it in this PR? It's for example merging all this https://github.com/iotaledger/iota-sdk/blob/b3f77d5a7a655741b3317a15997beef825062895/sdk/src/wallet/account/operations/syncing/addresses/output_ids/nft.rs#L29-L101

**Comment by thibault-martinez** [comment]
> > Is there nowhere we can actually use it already?
> 
> There is, should I include it in this PR? It's for example merging all this
> 
> https://github.com/iotaledger/iota-sdk/blob/b3f77d5a7a655741b3317a15997beef825062895/sdk/src/wallet/account/operations/syncing/addresses/output_ids/nft.rs#L29-L101

As you want, either in this PR or open an issue for later

**Review by kwek20 — APPROVED** [review]
lovely code cleanup 👌

**Review by DaughterOfMars — APPROVED** [review]
🙏

**Review by thibault-martinez — APPROVED** [review]
Very nice

**Pull request merged**